src/searching/src/bm_search.cairo

```diff
@@ -0,0 +1,72 @@
+// The Boyer-Moore string search algorithm
+use dict::Felt252DictTrait;
+
+// Find `pattern` in `text` and return the index of every match.
+/// * `text` - The text to search in.
+/// * `pattern` - The pattern to search for.
+/// # Returns
+/// * `Array<usize>` - The index of every match.
+fn bm_search(text: @ByteArray, pattern: @ByteArray) -> Array<usize> {
+    let mut positions: Array<usize> = array![];
+    let n = text.len();
+    let m = pattern.len();
+    if n == 0 || m == 0 || m > n {
+        return positions;
+    }
+
```

Comment:
add comments and add meaningful variable names please

Reply:
Fixed ;)